packages/webpack-import-meta-loader/webpack-import-meta-loader.js

```diff
@@ -24,9 +24,34 @@ module.exports = function(source) {
   const browserPath = toBrowserPath(relativePath);
 
   const fileName = this.resource.substring(this.resource.lastIndexOf(path.sep) + 1);
-  return source.replace(
-    regex,
-    () =>
-      `({ url: \`\${window.location.protocol}//\${window.location.host}/${browserPath}/${fileName}\` })`,
-  );
+
+  let found = false;
+  let rewrittenSource = source.replace(regex, () => {
+    found = true;
+    return `({ url: getAbsoluteUrl('${browserPath}/${fileName}') })`;
+  });
+
+  if (found) {
+    return `
+      function getAbsoluteUrl(relativeUrl) {
```

Comment:
If there are two `import.meta.url` in a file this function will be defined twice, right?

Reply:
No, it will be defined only once.

In the global regex replace above this code we set `found` to `true`. This indeed can happen multiple times if multiple `import.meta.url` are in the file. But afterwards, the extra `getAbsoluteUrl` function will be only added once per file.